### PR TITLE
WIP: Migrate documentation to source repository

### PR DESCRIPTION
**Pull request by jcfr**
See issue #4751

For now, readthedocs will automatically regenerate the documentation each time the branch [Slicer/Slicer@4751-migrate-documentation-to-source-repository](https://github.com/Slicer/Slicer/tree/4751-migrate-documentation-to-source-repository) is updated.

Initial set of changes in this branch was ported over from https://github.com/Slicer/SlicerGitSVNArchive/pull/686

Pending issues:
* [x] Fix integration of external images used in readthedocs
* [x] Update Copyright year
* [ ] Port over existing module wiki page using pandoc
* [ ] Add recommendation for compressing images + enable PR hook for checking size of files

Questions:
* Should we store figure binary sources (e.g pptx used in this commit: https://github.com/Slicer/Slicer/pull/4771/commits/ffd89df68c67c7ba584defc5cccbf5de5df4e319#diff-41e1283d0b9f7e13387615bc3aff4f5d) in the repository ? 
  * @jcfr: I would prefer not. Then, where do we store them ?

### CODE REVIEW

File: Docs/user_guide/getting_started.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+3D Slicer is built on a modular architecture. The Main Application GUI is divided into six components: the Application Menu Bar, the Application Toolbar, the Module GUI Panel, the Data Probe Panel, the 2D Slice Viewers, and the 3D Viewer.
+
+This section will introduce you to the basic functions on the main application's GUI. If you require detailed information, visit this page.
+
+
+Open 3D Slicer and load your own data or download sample data to explore. Go ahead and click around the user interface.
+
+
+.. image:: https://www.slicer.org/w/images/a/a6/LoadingData-16-06-06.png
```
Comment (lassoan) [inline Docs/user_guide/getting_started.rst]: URL has changed: https://www.slicer.org/w/img_auth.php/a/a6/LoadingData-16-06-06.png (and probably for all similar URLs)

File: Docs/user_guide/citing.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Gering D.T., Nabavi A., Kikinis R., Grimson W.E.L., Hata N., Everett P., Jolesz F.A., Wells III W.M. `An Integrated Visualization System for Surgical Planning and Guidance using Image Fusion and Interventional Imaging <http://www.spl.harvard.edu/publications/item/view/816>`_. Int Conf Med Image Comput Comput Assist Interv. 1999 Sep;2:809-19. 
+
+
+Individual Module
+-----------------
+
+To acknowledge individual modules:
+
+.. figure:: images/Acknowledgement-tab.png
```
Comment (lassoan) [inline Docs/user_guide/citing.rst]: We should probably move all images to https://discourse.slicer.org/c/dev/documentation-resources/26 and not upload these images here.

Comment (jcfr) [reply Docs/user_guide/citing.rst]: Good point, I totally forgot about this. I will do so.

Comment (jcfr) [reply Docs/user_guide/citing.rst]: But first, going on a :running_man: before the :sun_with_face: set

Comment (jcfr) [reply Docs/user_guide/citing.rst]: When you first drag and drop an image into  a discourse post, it is uploaded into a temporary location and is available using an url like this one (available doing right click `Copy Image Location` in the preview):

```
https://aws1.discourse-cdn.com/standard17/uploads/slicer/original/2X/d/sywbxlReSbohAlm7G5Hy8s47Osp.png
```
and it generates markdown like this one:

```markdown
![auto_complete|21x21](upload://sywbxlReSbohAlm7G5Hy8s47Osp.png)
```

Then, after saving the post, the markdown remains unchanged but the image is moved to an amazon S3 bucket and the url become:

``'
https://discourse-cloud-file-uploads.s3.dualstack.us-west-2.amazonaws.com/standard17/uploads/slicer/original/2X/c/c823cbb536a39bef882e6b75d10c3b9e8e92d931.png
```

where `discourse-cloud-file-uploads` is a bucket and `standard17/uploads/slicer/original/2X/c/c823cbb536a39bef882e6b75d10c3b9e8e92d931.png` is the key

Comment (jcfr) [reply Docs/user_guide/citing.rst]: Additionally, discourse server post-process the image to decrease it size (in this example from 316 bytes to 208 bytes) and the S3 key likely includes the SHA1 of the post-processed file (`c823cbb536a39bef882e6b75d10c3b9e8e92d931` in our example)

This means that if discourse backend switch to a different optimization of upload images  and re-process all the images ... all the link we use in readthedocs will be broken.

Comment (jcfr) [reply Docs/user_guide/citing.rst]: Looking at the code, it turns out that only `png` file are optimized:

https://github.com/discourse/discourse/blob/efd5fb665b0f478c0159b7dfb315dc0afa820c17/lib/upload_creator.rb#L334-L342

Comment (jcfr) [reply Docs/user_guide/citing.rst]: I am now able to recover the SHA1 by running `optipng`:

```
$ src/optipng/optipng -o2 /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png -keep -strip all

$ sha1sum /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png*
c823cbb536a39bef882e6b75d10c3b9e8e92d931  /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png
93070ab5f8ecf49ed4c01a98936834ea3b1762e8  /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png.bak
```

`c823cbb536a39bef882e6b75d10c3b9e8e92d931` is the SHA1 found in the URL

Comment (cpinter) [reply Docs/user_guide/citing.rst]: Thanks for investigating this, @jcfr ! So the bottom line is that we can use discourse as backend even after they decide to regenerate the optimized images?

Comment (jcfr) [reply Docs/user_guide/citing.rst]: > So the bottom line is that we can use discourse as backend even after they decide to regenerate the optimized images?

Indeed, now we understand the underlying process to generate the S3 keys, I am more confident to move forward with this approach. 

I also wrote a python scripts allowing to easily generate tables like the one displayed in this post:

https://discourse.slicer.org/t/segment-editor-module-resources/10841

**Comment by jcfr** [comment]
For future reference, to remove `./user_guide/images/module_data/SlicerDataFormats.pptx`from commit `DOC: Add "Data Loading and Saving" documentation`, the following commits:

* 0216b0b3e3 - DOC: Add support for including md document at root and include CONTRIBUTING info
* b36607d055 - DOC: Update Getting Started section to reference images on discourse
* 1c9ef1e377 - DOC: Fix slicer.util docstrings
* 3b65da6fb0 - DOC: Add "Segment Statistics" module documentation
* 57f5a3584f - DOC: Add "Pixel Medical" to "Commercial Partners" documentation
* be17acd228 - DOC: Add "DICOM Patcher" module documentation
* ef0d690652 - DOC: Add "Data Loading and Saving" documentation

have been forced pushed with

* fc8c5c648b - DOC: Add support for including md document at root and include CONTRIBUTING info
* bb536facba - DOC: Update Getting Started section to reference images on discourse
* ac24a9cad1 - DOC: Fix slicer.util docstrings
* 105d5f77c7 - DOC: Add "Segment Statistics" module documentation
* f7a1dc1288 - DOC: Add "Pixel Medical" to "Commercial Partners" documentation
* 565ca41511 - DOC: Add "DICOM Patcher" module documentation
* 77680b4b78 - DOC: Add "Data Loading and Saving" documentation

**Pull request merged**